### ISSUE TITLE
Can't instantiate Interval with name attribute

### ISSUE THREAD

**Issue by clintval** [daler/pybedtools#217]
I can reproduce the following on v0.7.7 and the bleeding edge v0.7.9 with Python 3:

```python
>>> from pybedtools import Interval
>>> interval = Interval('chr1', 2, 10, name='test', strand='+')
>>> print(interval)
chr1	2	10	test	.	+
```

Yet when trying to print the `name` attribute or key only `None` is returned:

```python
>>> print(interval.name)
None
>>> print(interval['name'])
None
```
Interestingly I can access the name by `interval.fields` as well as the `strand` attribute and key:

```python
>>> print(interval.fields)
['chr1', '2', '10', 'test', '.', '+']
>>> print(interval.strand)
+
>>> print(interval['strand'])
+
```
Happy to look into the problem but figured I would start with asking if anyone had an idea on why the `name` attribute is not being accessed correctly after it is set? Can you reproduce? Thanks!

**Comment by daler** [comment]
Thanks for reporting. I can reproduce, it seems to be that the `file_type` attribute is not being set properly upon direct instantiation of an `Interval` object. Setting the `file_type` to `"bed"` manually fixes the issue, or using the more complex `create_interval_from_list()`:

```python
import pybedtools

# the doctest at
# https://daler.github.io/pybedtools/intervals.html#common-interval-attributes
# passes, so let's start with that
x = pybedtools.example_bedtool('a.bed')[0]
print(x)
assert(x.name == 'feature1')

# construct another interval using the same fields
y = pybedtools.Interval(
    x.fields[0],
    int(x.fields[1]),
    int(x.fields[2]),
    x.fields[3],
    x.fields[4],
    x.fields[5]
)

# and using create_interval_from_list, which many internal functions use:
z = pybedtools.create_interval_from_list(x.fields)

# They are identical in all meaningful ways . . . .
assert x.fields == y.fields == z.fields
assert str(x) == str(y) == str(z) == 'chr1\t1\t100\tfeature1\t0\t+\n'
assert type(x) == type(y) == type(z) == pybedtools.Interval
assert x.chrom == y.chrom == z.chrom == 'chr1'
assert x.start == y.start == z.start == 1
assert x.stop == y.stop == z.stop == 100
assert x.strand == y.strand == z.strand == '+'
assert x.score == y.score == z.score == '0'

# Except `name`:
try:
    assert x.name == y.name == z.name == 'feature1'
except AssertionError:
    print('name for `x` is: "{}"'.format(x.name))
    print('name for `y` is: "{}"'.format(y.name))
    print('name for `z` is: "{}"'.format(z.name))

# based on
# https://github.com/daler/pybedtools/blob/master/pybedtools/cbedtools.pyx#L403,
# the field for name is determined by the file_type.
try:
    assert x.file_type == y.file_type == z.file_type == 'bed'
except AssertionError:
    print('file type for `x` is: "{}"'.format(x.file_type))
    print('file type for `y` is: "{}"'.format(y.file_type))
    print('file type for `z` is: "{}"'.format(z.file_type))

# manually setting the file_type allows the test to pass
y.file_type = 'bed'
assert x.name == y.name == z.name == 'feature1'
```

I can't see what the problem is after a quick look, but if you wanted to poke around that would be great. You'll probably want to start looking at:

- https://github.com/daler/pybedtools/blob/master/pybedtools/cbedtools.pyx#L403
- https://github.com/daler/pybedtools/blob/master/src/bedFile.h#L199
- https://github.com/daler/pybedtools/blob/master/pybedtools/cbedtools.pyx#L582

**Comment by clintval** [comment]
Thanks for the attention @daler . I did notice that the `name` attribute seemed tethered to the `file_type` logic regardless of how if you instantiated `name` directly. I can spend a little more time on this later in the week and push a PR.

**Comment by daler** [comment]
Addressed in #264
